packages/protect/src/lib/protect.test.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ *
+ * Copyright © 2025 Ping Identity Corporation. All right reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ *
+ */
+
+import FRStep from '@forgerock/javascript-sdk/src/fr-auth/fr-step';
+import { protect } from './protect.js';
+import {
+  noProtectType,
+  standardPingProtectEvaluationStep,
+  standardPingProtectInitializeStep,
+} from './protect.mock.data.js';
+import { ProtectConfig, Protect } from './protect.types.js';
+import { CallbackType, HiddenValueCallback } from '@forgerock/javascript-sdk';
+
+describe('protect', () => {
+  const config: ProtectConfig = {
+    envId: '12345',
+    consoleLogEnabled: true,
+    deviceAttributesToIgnore: ['userAgent'],
+    lazyMetadata: false,
+    behavioralDataCollection: true,
+    deviceKeyRsyncIntervals: 14,
+    enableTrust: false,
+    disableTags: false,
+    disableHub: false,
+  };
+  let protectAPI: Protect | undefined;
+
+  async function getProtect(options: ProtectConfig): Promise<Protect> {
+    const protectAPI: Protect = await protect(options);
+    expect(protectAPI).toBeDefined();
+    assertType<Protect>(protectAPI);
+    return protectAPI;
+  }
+
+  beforeEach(async () => {
+    vi.doMock('./signals-sdk.js', () => {
+      return {
+        default: {
+          init: vi.fn(),
+          getData: vi.fn(),
+          pauseBehavioralData: vi.fn(),
+          resumeBehavioralData: vi.fn(),
+        },
+      };
+    });
+
+    if (typeof window === 'undefined') {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      global.window = {} as any;
+    }
+
+    window._pingOneSignals = {
+      init: vi.fn().mockResolvedValue(undefined),
+      getData: vi.fn().mockResolvedValue('mocked-data'),
+      pauseBehavioralData: vi.fn(),
+      resumeBehavioralData: vi.fn(),
+    };
+
+    protectAPI = await getProtect(config);
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    delete (window as any)._pingOneSignals;
+  });
+
+  it('should be defined', () => {
+    expect(protect).toBeDefined();
+  });
```

Comment:
Anytime I see `beforeEach` and `afterEach`, I start to wonder if we are designing what's tested correctly, or we're designing the tests right. I know we may be able to reduce some repetition by doing this, but I'd much prefer to see `protect` initialized in each `it` as well as the mocks. That way, we don't rely on side-effects or shared values.